sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -47,6 +50,26 @@ def __init__(self, gcs_staging=None, namespace=None):
     from ._gcs_helper import GCSHelper
     GCSHelper.create_gcs_bucket_if_not_exist(gcs_bucket)
 
+    # Configure the GCR image tag
+    # Kubeflow jupyter notebook is injected with a variable as an ID
+    if gcr_image_tag is None:
+      KF_NOTEBOOK_ENV = 'NB_PREFIX'
+      nb_id = str(uuid.uuid4())
+      try:
+        if KF_NOTEBOOK_ENV in os.environ.keys():
+          nb_id = os.environ[KF_NOTEBOOK_ENV]
+        else:
+          nb_id = self._get_instance_id()
+      except:
+        pass
+      nb_id = nb_id.replace('/', '-').strip('-')
+      self._gcr_image_tag = os.path.join('gcr.io', self._get_project_id(), nb_id, GCR_DEFAULT_IMAGE_SUFFIX)
```

Comment:
will container build function be called from a local environment? 
wonder if _get_project_id() work when the call is from there?

Reply:
For local environment, it will raise an exception and users need to pass the gcr_image himself.

Reply:
Does user have to provide the GCR path? Or it can be any docker image registry?
I'd recommend we think this part more throughout. 
- What is the long term platform independent interface
- How the interface incorporate with GCP
- What's the appropriate default value

Reply:
The current containerbuilder only supports GCR.  https://github.com/kubeflow/pipelines/issues/1700 tracks the support for other registries. 


> Does user have to provide the GCR path? Or it can be any docker image registry?
> I'd recommend we think this part more throughout.
> 
> * What is the long term platform independent interface
This interface is internal, the user facing interface is driven by Alexey. @Ark-kun 
I would imagine the container builder related user-facing platform independent interface for the component builder to be nothing but a directory that contains the required files and the dockerfile name.  

> * How the interface incorporate with GCP
Not quite understand what the question means. Would you elaborate?
This interface currently depends on the GCS/GCR services of GCP. In case users run the job inside the GCP, it extracts the metadata information such as project_id, instance_id. Otherwise, the users need to specify the default GCS/GCR location from the local environment.

> * What's the appropriate default value
Assuming the current support is GCR. This PR proposes 'gcr.io', self._get_project_id(), nb_id, GCR_DEFAULT_IMAGE_SUFFIX) that separates the image tag for each notebook environment. In KF notebook, it is the nb_prefix, which consists of the namespace and the notebook name. In AI Platform notebook, it is the instance id. WDYT?